File: consecutive_number/consecutive.js
const readline = require('readline');
const rl = readline.createInterface({ input: process.stdin, output: process.stdout });

function findConsecutiveNumbers(targetSum) {
  const resultList = [];
  let start = 1;
  let end = 1;
  let currentSum = start;

  while (start <= Math.floor(targetSum / 2)) {
    if (currentSum === targetSum) {
      if (start !== end) {  // Avoid single numbers
        const sequence = Array.from({ length: end - start + 1 }, (_, i) => start + i);
        resultList.push(sequence);
      }
      end += 1;
      currentSum += end;
    } else if (currentSum < targetSum) {
      end += 1;
      currentSum += end;
    } else {
      currentSum -= start;
      start += 1;
    }
  }

  return resultList;
}

rl.question("input: ", function (user_input) {
  const result = findConsecutiveNumbers(parseInt(user_input));
  const resultStrings = result.map(sequence => `[${sequence.join(', ')}]`);
  console.log(`output: ${resultStrings.join(', ')}`);
  rl.close();
});

File: consecutive_number/consecutive.py
def find_consecutive_numbers(target_sum):
    result_list = []
    start = 1
    end = 1
    current_sum = start

    while start <= target_sum // 2:
        if current_sum == target_sum:
            if start != end:  # Avoid single numbers
                result_list.append(list(range(start, end + 1)))
            end += 1
            current_sum += end
        elif current_sum < target_sum:
            end += 1
            current_sum += end
        else:
            current_sum -= start
            start += 1

    return result_list

user_input = input("input\t: ")
result = find_consecutive_numbers(int(user_input))
result = [str(r) for r in result]

print(f"output\t: " + ", ".join(result))